### ISSUE TITLE
Show expected and actual value for string equality

### ISSUE THREAD

**Issue by webmaster128** [jtenner/as-pect#188]
When I compare two strings, the debugging output is not very helpful at the moment:

```ts
    it("strings are equal", () => {
      expect("a").toStrictEqual("b");
    });
```

prints

```
    ❌ strings are equal - 
      [Expected]: Reference Value
      [Actual]  : Reference Value
```

Would it be possible to create an output that helps understanding the problem (i.e. show expected and actual string)?

**Comment by jtenner** [comment]
See that's definitely not right. I can't check it out tonight, but I will open a pull request tomorrow to investigate the problem.

**Comment by webmaster128** [comment]
Thanks! Just in case it matters: I use `"--runtime": ["stub"],`

**Comment by jtenner** [comment]
Ah yes. That might be why.

I'll try to see what's going on.

**Comment by jtenner** [comment]
This won't matter. Indeed, the `toStrictEqual()` function calls the `blockComparison` function.

Which leads to this...

https://github.com/jtenner/as-pect/blob/master/packages/assembly/assembly/internal/comparison/blockComparison.ts#L43

In this "block comparison," lines 43 through 62 can be replaced with exactly two lines of code which would correctly report the strings and any other kinds of memory with variable heap allocation sizes.

In short, I made an oopsie.

Strangely, it's valid to use:

```ts
expect("a").toBe("b");
```

This would have reported the strings correctly and given you the same result you desired.

**Comment by jtenner** [comment]
Ironically, the output for actual and expected values should be a bit more clear regardless:

```
❌ assembly/__tests__/example.spec.ts Pass: 0 / 1 Todo: 0 Time: 5.182ms
  Failed: stuff
    ❌ a -
      [Expected]: b
      [Actual]  : a
```

And it might be reasonable to do something like:

```
❌ assembly/__tests__/example.spec.ts Pass: 0 / 1 Todo: 0 Time: 5.182ms
  Failed: stuff
    ❌ a -
      [Expected]: String: {blueHighlight "b"}
      [Actual]  : String: {blueHighlight "a"}

**Comment by webmaster128** [comment]
Ah, I did not think of `.toBe`. I try to stick with some kind of equality matcher when I don't explicitely want to test object identity.

**Comment by jtenner** [comment]
I should be pulling the request that fixes this issue soon

**Comment by jtenner** [comment]
@webmaster128 I wanted to mention that strings are immutable in nature, but using `.toStrictEqual` is a valid way to compare them, even if the functions that compare strings get used the same way.
